### ISSUE TITLE
Node 3.1.0 starts in different modes, eventually gets stuck.

### ISSUE THREAD

**Issue by lkjhkj** [openethereum/openethereum#148]
I am yet to fully sync the 3.1.0 node on Ubuntu 18, but I have no idea why it sometime starts as following:
![2020-12-04-083306_1365x299_scrot](https://user-images.githubusercontent.com/62512125/101125920-6f9c8180-360b-11eb-997c-2bba7b122a98.png)
and sometimes like this:
![2020-12-04-083255_1137x138_scrot](https://user-images.githubusercontent.com/62512125/101125930-76c38f80-360b-11eb-8ac8-5998a1f4b491.png)
Why is it so? What's the difference? When either of these are finished to 100% I may consider my node fully synced?

The problem is either of these eventually gets stuck (drops to `0 blk/s` or stuck at some snapshot repeating the same number.)

# UPDATE
Luckily, I could reach `Syncing snapshot 5706/5706` this time, and now it counts backwards with `Block restoration`:
![2020-12-04-113623_1176x563_scrot](https://user-images.githubusercontent.com/62512125/101141184-fc076e00-3624-11eb-9c87-cded50a509d1.png)
Will it go down to the lowest block? And then what?

# UPDATE 2
So the process killed itself. Great job, guys!

**Comment by SomethingUseful** [comment]
Waited 4 weeks to sync (100k blocks to go) then death...
`2020-12-29 17:47:02 UTC main WARN kvdb_rocksdb  DB corrupted: Invalid argument: You have to open all column families. Column families not opened: col4
, col6, col1, col3, col0, col2, attempting repair`

**Comment by hhanh00** [comment]
I have never been able to sync. It always hangs at one point when syncing snapshots.

```
2021-01-21 22:13:18  Syncing snapshot 4339/6525  #11465000   26/50 peers   1 MiB chain 0 bytes queue  RPC:  0 conn,    0 req/s,    0 µs
2021-01-21 22:13:23  Syncing snapshot 4339/6525  #11465000   26/50 peers   1 MiB chain 0 bytes queue  RPC:  0 conn,    0 req/s,    0 µs
2021-01-21 22:13:28  Syncing snapshot 4339/6525  #11465000   26/50 peers   1 MiB chain 0 bytes queue  RPC:  0 conn,    0 req/s,    0 µs
...
```

Can openethereum warp sync with other non openethereum peers?

**Comment by rakita** [comment]
warp sync is specific OpenEthereum feature. There was a syncing bug that is fixed in v3.1.1
